### DISCUSSION TITLE
How to detect failed user login events?

### DISCUSSION THREAD

**Discussion by gabriel-hoff-vectorsolutions** [zitadel/zitadel#7830] [Q&A]
Hello!

I'm trying to understand how I can proactively detect when users cannot log in for any reason. Specifically, I'm looking to detect when:

- Applications are misconfigured (e.g., redirect URI not included in the application configuration, grant type is not allowed, etc.)
- Actions not allowed to fail are failing
- SAML identity providers are misconfigured

I was expecting that the events or failed events functionality would provide me with something that I could use for that, but I've tried reproducing the scenarios above, and none of them seem to generate any event.

Am I misunderstanding the purpose of events and failed events? Is proactive detection of login failures something not supported by Zitadel at the moment?

**Comment by hifabienne** [comment]
Hei @vs-gtadeu 
The failed events are more for internal use. As ZITADEL works on eventsourcing technology, we do compute the readmodels. If something does fail in computing the readmodels an event is added in the failed events.

I guess the easiest way would be to setup an end to end test that implements your use case of authenticating a user.
@adlerhurst @livio-a any better ideas?

**Reply by gabriel-hoff-vectorsolutions** [reply]
> As ZITADEL works on eventsourcing technology, we do compute the readmodels. If something does fail in computing the readmodels an event is added in the failed events.

That makes a lot of sense. Thanks for clarifying!

> I guess the easiest way would be to setup an end to end test that implements your use case of authenticating a user.

Yes, and we were already planning to implement those. But unfortunately, synthetic/end-to-end testing doesn't provide a way to detect if real users are having trouble logging in, so we hope to find a way to have both observability approaches implemented.

**Reply by livio-a** [reply]
I guess this is related to business metrics: https://github.com/zitadel/zitadel/issues/4458
And from what i got, it's mainly an OPS topic so ideally you would want a promQL API or similar to detect operational issues, correct?

**Reply by gabriel-hoff-vectorsolutions** [reply]
Yeah, this is mainly focused on detecting operational issues. Any query language would work for us. FWIW, we use Datadog for observability, so if we could get those metrics along with https://github.com/zitadel/zitadel/issues/6182, that would be fantastic!

**Reply by gabriel-hoff-vectorsolutions** [reply]
@livio-a, any news on this?
